### ISSUE TITLE
Bug in code for finger tutorial

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4420]
|<img alt="grifaton's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| grifaton reported|
|-|-|
|Trac ID|trac#4420|
|Type|defect|
|Created|2010-05-01 14:50:48Z|

There is a bug in the second section of the finger tutorial, under the heading _Read Status File_.

The `_read` method of a `FingerService` object is called twice - once in `__init__` and once in `startService`.  This means it is called twice every thirty seconds.  This also means it is only stopped once, in the call to `stopService`.

I think the call to `_read` should be removed from `__init__`.

Attachments:

* [finger-doubleread.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/6b8/6b8bb507f6533a10c6d641b81a60be15e7953c90/ad71f7ba9ae103a51d22da79ff36b1e16a3c6416.patch) (474 bytes) - added by exarkun on 2010-05-01 18:54:09Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__4420 4420
type__defect defect
reporter__grifaton grifaton
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1272725448000000 1272725448000000
changetime__1275271670000000 1275271670000000
version__None None
owner__ 
cc__rwall
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun removed owner|
|-|-|

Attached patch fixes the _one_ (**one**) tac file which has this problem.  It seems that after `startService` and `stopService` are introduced, they are promptly forgotten about.  As far as I can tell only `finger14.tac` needs to be changed here.

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_2"></a>|@wallrj set owner to grifaton|
|-|-|

Replying to [exarkun](#note_1):
> Attached patch fixes the _one_ (**one**) tac file which has this problem.  It seems that after `startService` and `stopService` are introduced, they are promptly forgotten about.  As far as I can tell only `finger14.tac` needs to be changed here.

I don't agree. It seems to me that start,stopService are the correct places to control the repeating call to re-read the users file.

I think {start,stop}Service methods should be continued throughout the series of tacs from finger14.tac onwards.

If you prefer to start the _read call from __init__,  then do so in in this patch too and remove the start,stopService methods from finger14.tac.

Also the patch doesn't include NewsFiles.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

> I think {start,stop}Service methods should be continued throughout the series of tacs from finger14.tac onwards. 

That sounds like a larger issue than the one reported here.  If someone wants to take it on, they can. :)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="mithrandi's avatar" src="https://avatars.githubusercontent.com/u/856430?s=50" width="50" height="50">](https://github.com/mithrandi)<a name="note_5"></a>|@mithrandi set owner to @exarkun|
|-|-|

Looks good, please merge (although I guess you need a news file first).

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set status to `closed`|
|-|-|

(In [29115]) Fix double loop start in finger14.tac

Author: exarkun
Reviewer: mithrandi
Fixes: [#4420](https://github.com/twisted/twisted/issues/4420)

finger14.tac previously started two callLater loops to re-read its users file,
once using the service interface and once in a plain `__init__`.  The second
of these was redundant, so get rid of it.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
